pkgs/servers/sql/postgresql/README.md

```diff
@@ -0,0 +1,22 @@
+# PostgreSQL versioning
+
+PostgreSQL's versioning policy is described [here](https://www.postgresql.org/support/versioning/). TLDR:
+
+- Each major version is supported for 5 years.
+- Every three months there will be a new minor release, containing bug and security fixes.
+- For criticial/security fixes there could be more minor releases inbetween. This happens *very* infrequently.
+- After five years, a final minor version is released. This usually happens in early November.
+- After that a version is considered end-of-life (EOL).
+- Around February each year is the first time an EOL-release will not have received regular updates anymore.
+
+Technically, we'd not want to have EOL'ed packages in a stable NixOS release, which is to be supported until one month after the previous release. Thus, with NixOS' release schedule in May and November, the oldest PostgreSQL version in nixpkgs would have to be supported until December. It could be argued that a soon-to-be-EOL-ed version should thus be removed in May for the .05 release already.
```

Comment:
There are two aspects I'd like to see reflected here:

* with the last release coming in November, we technically have coverage until February because vulnerabilities are disclosed for the other versions in February and not earlier. For me this was the primary reason why I agreed IIRC.
* if there are "emergency releases" before and within the lifetime of the old release (in our case 24.05), then we'll mark the package as insecure unless we can _trivially_ patch the issue.

Reply:
> This is addressed in the very last paragraph already. I can rephrase that to use "trivially".

Eh yes, sorry. I define "trivially" as "patch applies without conflicts" here and <=5 patches to apply. Everything else seems a little too error-prone to me then. WDYT?